Modal: add `size` prop to support a selection of preset modal sizes

chad1008 opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
Implements a new prop for `Modal` that constrains the content width to one of a selection of preset sizes.

## Why?
Consumers were finding they often needed to manually set a max width on the child elements they provided to `Modal` to prevent them from pushing `Modal`'s size to the full with of the viewport.

## How?
The optional new prop accepts values of `small`, `medium`, and `large`. If provided, the width of the `Modal`'s contents will be limited to `300px`, `600px`, or `900px`, respectively. (Note: these are just my initial values, and can most certainly be changed as needed when this PR is reviewed).

The value provided to the prop is used to add an additional class name to the relevant DOM element, but this class is only added if `isFullscreen` is `false`, so full screen modals will not be affected. Further, the styles implementing this change are limited to larger screens, so the automatically full screen `Modal` rendering on mobile devices should not be affected either.


- [x] TODO: rebase and refactor changes in #54518 to use the new APIs proposed in this PR
---

dropping a ping for some folks who may want to weigh in on any design considerations for this one: @jordesign @koster @SaxonF

## Testing Instructions
`npm run storybook:dev`

- Open `Modal`'s new `With contentWidth: small` story and test the `contentWidth` control.
- Confirm that both `isFullScreen` behaves as expected with the new prop applied
- Confirm smaller viewports still render the modal in full screen, regardless of the new prop's value

### Testing Instructions for Keyboard
n/a

chad1008 committed:
add `contentWidth` prop, styles, and docs

brookewp commented on packages/components/src/modal/stories/index.story.tsx:
Since this is the 'Default' story, this change made me think that the `medium` width was the default. I don't know if it's a problem, but would it be worth adding a new story for content width? Then, one of the new sizes, large or small, could show the differences from the default.

brookewp replied:
Another change to the default story is when setting `isFullScreen`, the inline style overrode the `max-width`. But with your changes, it appears to behave the way I would have expected from `isFullScreen`: 

**trunk:**

<img width="1391" alt="Screenshot 2023-09-14 at 3 57 38 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/971c906f-9fc1-4c40-af6a-a0648731ce8f">

**this branch:**

<img width="1385" alt="Screenshot 2023-09-14 at 3 58 00 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/de2c5335-3ce9-48e4-b7d3-337bbde4e76f">

Which I realize conflicts with what I said above about the default story. So I wonder if it's more important to keep the story the same as before or to make sure the `isFullScreen` behavior isn't overridden? 🤔

chad1008 replied:
Oh, that's interesting. I hadn't noticed that the `style` prop setting a `maxWidth` overrode `isFullScreen`'s behavior that way! It makes sense now that I'm thinking about it, but it does seem suboptimal that a consumer could easily (and probably accidentally) negate the `isFullScreen` prop this way.

Might be worth addressing in a followup PR. We could put an `!important` onto `isFullScreen`'s `max-width: none` to make sure it can always go full width. I'll hate it, because I despise `!important`, but perhaps a necessary evil in this case? 🤔 

Either way, I was on the fence about a dedicated story for the new prop, and you're right... the default story using `medium` does imply a little that that's the default setting. I'll throw a new story in real quick!

ciampo replied:
I think the current setup in this PR makes sense — not having any inline style override will show how the component really behaves in isolation.

Also, I wouldn't worry too much about adding `!important`. Folks overriding styles should expect the chance that the component may not behave as expected.

brookewp commented on packages/components/src/modal/README.md:
```diff
@@ -215,6 +215,12 @@ This property when set to `true` will render a full screen modal.
+#### `contentWidth`: `'small' | 'medium' | 'large'`
+
+If this property is added, it will constrain the `max-width` of the modal's contents, preventing it from growing too wide. This prop only applies when `isFullScreen` is `false`.
+
```
Missing `Required: No`

chad1008 replied:
Yeah I was... testing you. See if you noticed. Yeah. You nailed it, congrats. 😆

brookewp commented on packages/components/src/modal/index.tsx:
```diff
@@ -85,6 +86,8 @@ function UnforwardedModal(
 	const [ hasScrolledContent, setHasScrolledContent ] = useState( false );
 	const [ hasScrollableContent, setHasScrollableContent ] = useState( false );
 
+	const contentWidthClass = contentWidth ? `width-${ contentWidth }` : '';
```
I've noticed that additional CSS classes on components usually start with `is` or `has`. Like `is-full-screen` and `has-icon`: 

<img width="515" alt="Screenshot 2023-09-14 at 3 48 18 PM" src="https://github.com/WordPress/gutenberg/assets/35543432/6e296f09-1743-4a15-9840-8de90c347536">

I'm not sure if there's documentation about it, but it stood out to me as different from the norm. Maybe it could be some combination like `has-${ contentWidth }-width` or `is-${ contentWidth }-small` instead?

chad1008 replied:
Good observation. `has-${ contentWidth }-width` FTW!

chad1008 committed:
update classNames

chad1008 committed:
add contentWidth story

chad1008 committed:
add `required: no` to README

ciampo reviewed (commented):
It would be good to get some input from @WordPress/gutenberg-design :

- is the proposed approach compatible with your needs (ie. setting a max width) ?
- do you have any advice on the number of set widths (for now we added 3 options, small / medium / large) and the corresponding px widths (300/600/900) ?
- do you have any advice on what is the expected behaviour of the component:
  - when the viewport's width is _lower_ than the selected max width
  - when the `isFullScreen` prop is `true` (which makes the modal go full screen on mobile screens)

jameskoster commented:
> is the proposed approach compatible with your needs (ie. setting a max width) ?

That seems reasonable to me, we might want to consider a min-width too.

> do you have any advice on the number of set widths

Perhaps we'll want to align these with general content widths that emerge out of https://github.com/WordPress/gutenberg/issues/53322. For now using values from the 4px baseline is probably adequate. Maybe:

* Small – 384px. This can work well for `confirmDialog`s, renames, and single-action modals
* Medium – 512px. Welcome guides, Keyboard shortcuts.
* Large – 832px. Template creation, editor preferences.
* Fill – stretched. Media library, pattern explorer, etc.

> when the viewport's width is lower than the selected max width

I may be missing something but I don't think I'd expect any special behavior in this case? The modal would just fill the width (accounting any margin or padding on the container).

We _might_ consider positioning modals towards the bottom of the viewport on mobile widths, similar to sheets on mobile OS's e.g.:

<img width="375" alt="Mobile" src="https://github.com/WordPress/gutenberg/assets/846565/25ba225f-85b0-4ba7-bbc0-6a2538b7021e">

I acknowledge that's probably an optimisation to explore separately. 

> when the isFullScreen prop is true (which makes the modal go full screen on mobile screens)

This doesn't only affect mobile, it also stretches the modal to fill the viewport on desktop, right? Should this be deprecated in favor of a "fill" `size`? It would be a bit strange imo to allow a `small` + `isFullScreen` combo.

richtabor commented:
Just a note. Is it intentional that command palette width is omitted from this, or is the idea that it conforms to one of these sizes?

It sits at 420px right now. If the latter, 512px seems quite wide for the palette imo. The larger it is, the more it blocks the context of where you are/what you're doing. 

<img width="505" alt="CleanShot 2023-09-18 at 16 14 21" src="https://github.com/WordPress/gutenberg/assets/1813435/3c50955f-cd9f-4a4e-b1be-9cc4fa83ce28">

SaxonF commented:
+1 on eventually standardising width values.

One tiny detail. I wonder if shorter `sm`, `md`, `lg` format might be more [familiar](https://tailwindcss.com/docs/max-width) to folks and slightly more scalable e.g. xs, xl, 2xl etc

ciampo commented:
@jameskoster 

> I may be missing something but I don't think I'd expect any special behavior in this case? The modal would just fill the width (accounting any margin or padding on the container).

Sounds good — I was indeed wondering if there should be some margin around the modal to that some portion of the backdrop is always revealed

> We might consider positioning modals towards the bottom of the viewport on mobile widths, similar to sheets on mobile OS's

FWIW, I've also seen examples where a Modal becomes a "bottom drawer" on mobile screens. But that's a separate conversation.

> This doesn't only affect mobile, it also stretches the modal to fill the viewport on desktop, right? Should this be deprecated in favor of a "fill" size? It would be a bit strange imo to allow a small + isFullScreen combo.

I must have gotten confused by [the storybook example](https://wordpress.github.io/gutenberg/?path=/story/components-modal--default)), where we the arbitrary `max-width` that we set via inline styles is interfering with full screen styles.

I also agree that a better solution could be to deprecate the `isFullScreen` prop in favour of one extra option for the new prop that we're adding in this PR 👌 

----

@richtabor 

> Is it intentional that command palette width is omitted from this, or is the idea that it conforms to one of these sizes?

I'll let the rest of the design team answer this, but let's keep in mind that the aim of this new prop is to give a few presets for the `Modal` width, but consumers of the component should be able to specify any custom `width` via standard CSS as always.

----

@SaxonF 

> One tiny detail. I wonder if shorter sm, md, lg format might be more [familiar](https://tailwindcss.com/docs/max-width) to folks and slightly more scalable e.g. xs, xl, 2xl etc

I don't have a strong opinion on this, but we should mention that historically components in this package used the full word when referring to sizing (ie. `'small'`, `'large'` etc.).

ciampo commented on packages/components/src/modal/stories/index.story.tsx:
```diff
@@ -91,6 +87,7 @@ const Template: StoryFn< typeof Modal > = ( { onRequestClose, ...args } ) => {
 export const Default: StoryFn< typeof Modal > = Template.bind( {} );
 Default.args = {
 	title: 'Title',
+	style: { maxWidth: '600px' },
```
As noted in [this comment](https://github.com/WordPress/gutenberg/pull/54471#issuecomment-1725088405), while this `max-width` rule makes the modal look more proportioned, it interferes with the `isFullScreen` prop (and with the new `contentWidth` prop). Probably best to remove it.

chad1008 committed:
remove `maxWidth` from default story

chad1008 committed:
add `contentWidth: fill` to replace `isFullScreen`

chad1008 committed:
update widths to more consistent values

chad1008 commented:
Thank you, everyone, for the input! I've pushed some changes to incorporate the above ideas. Summarizing the new bits:

- removed the `max-width` from the default story, so it better represents a vanilla `Modal` and doesn't interfere with the new `contentWidth` control.
- added a `fill` option for the new prop. This will mirror the functionality of the current `isFullScreen` prop. Note, the `isFullScreen` prop still works as it always did.
- added a deprecation warning for `isFullScreen` - @ciampo, this is the first prop I'm deprecating, please feel free to double check I'm going about it properly 😄
- updated the small/medium/large widths to the values [suggested](https://github.com/WordPress/gutenberg/pull/54471#issuecomment-1723818809) by @jameskoster (thank for those btw, I like the feel of them much better, and it's good to know there's consistency). I also added a note linking back to that comment for context on where those numbers came from.

jameskoster commented:
I wonder if medium should be the default instead of a dedicated size in the scale? Ideally consumers are creating modals that are consistent with those in core 99% of the time, and custom sizes are the exception. The current setup might promote custom sizing more than we like?

jameskoster commented on packages/components/src/modal/style.scss:
```diff
@@ -146,4 +146,25 @@
+		// The following widths were selected to align with existing baselines
+		// found elsewhere in the editor.
+		// See https://github.com/WordPress/gutenberg/pull/54471#issuecomment-1723818809
+		&.has-width-small {
+			max-width: 384px;
+		}
+		&.has-width-medium {
+			max-width: 512px;
+		}
+		&.has-width-large {
+			max-width: 832px;
+		}
```
Should these values be abstracted into variables? It may be useful for container widths (see https://github.com/WordPress/gutenberg/issues/53322).

ciampo replied:
> Should these values be abstracted into variables?

If we want these values to be shared across the whole project (not just modal / components), we could for now add the to the base styles package (knowing that they could be later be moved to whatever the centralised place for style variables is going to be)

ciampo commented on packages/components/src/modal/index.tsx:
```diff
@@ -85,6 +87,19 @@ function UnforwardedModal(
+			version: '6.7',
+			hint: 'Set the `contentWidth` prop to `fill` instead.',
+		} );
+	}
```
Even if we continued down the path of replacing `isFullScreen` with `contentWith="fill"`, to keep things more iterative and cleaner, we should remove the deprecation notice from this PR. Instead, we should plan a couple of follow-ups:

- first, we should refactor all existing usages of the `isFullScreen` prop to using `contentWidth="fill"`
- only after that, we should deprecate the prop.

chad1008 replied:
Got it, that makes a lot of sense. Thanks for clarifying. Deprecation notice removed, I'll address in followup.

ciampo commented on packages/components/src/modal/README.md:
```diff
@@ -215,6 +215,14 @@ This property when set to `true` will render a full screen modal.
+
+-   Required: No
+
+Note: `Modal`'s width can also be controlled by adjusting the width of the modal's contents, or using the `style` prop to set a specific `max-width`.
```
```suggestion
Note: `Modal`'s width can also be controlled by adjusting the width of the modal's contents via CSS.
```

ciampo commented on packages/components/src/modal/style.scss:
```diff
@@ -146,4 +146,25 @@
 		outline-offset: -2px;
 	}
+
+	@include break-small() {
```
Is there a reason why these styles are being applied to `components-modal__content` instead of `components-modal__frame` ?

I think it would be good to have full screen styles and "content-width" styles applied to the same element and in the same place in the styles file, since they all affect the same properties (width and max-width)

chad1008 replied:
That makes a lot of sense. I think I was overly focused on the `contentWidth` naming that got suggested in our initial discussions prior to this PR being created, so I dropped them into the `components-modal__content`. Fixed!

ciampo commented on packages/components/src/modal/style.scss:
```diff
@@ -146,4 +146,25 @@
+	@include break-small() {
+		&.has-width-small,
+		.has-width-medium,
+		.has-width-large {
```
I believe that the medium and large rules are missing the `&.` in front

```suggestion
		&.has-width-small,
		&.has-width-medium,
		&.has-width-large {
```

ciampo reviewed (commented):
As I reviewed this PR, I had a thought.

I'm not sure that it makes sense to replace `isFullScreen` with `contentWidth="fill"`. That's because making the modal full screen doesn't only affect the width, but also the height of the component — and therefore using a prop called `contentWidth` doesn't seem appropriate.

Also, I don't really like the `content` part either — after all, we're setting the size of the whole modal, not just its contents.

Therefore, I propose two alternatives:

- we keep the `isFullScreen` prop around (no deprecation), and we add a new prop (`contentWidth`, or a better name if we can find one). When the `isFullScreen` prop is set, the new prop gets ignored.
- we find a new name for the `contentWidth` prop (one that doesn't only refer to the width, and that doesn't only refer to the modal contents), so that we can effectively have it replace the `isFullScreen` prop

What do folks think?

chad1008 committed:
move new modal size classNames to modal frame instead of content

chad1008 committed:
README description update

chad1008 committed:
remove deprecation warning until existing implementations are migrated

chad1008 commented:
Thanks @jameskoster and @ciampo. After those helpful comments I'd propose the following:

- rename `contentWidth` to something like `modalSize` or just `size`.
- `size="fill"` will replace `isFullScreen`, which will eventually be deprecated.
- set `size="medium"` to be the default value.

This way we have only one prop handling the sizing of the modal, and it has a name more accurate name. Thoughts?

jameskoster commented:
I think the thing to avoid is contradictory configurations like size=small & isFullScreen, so that seems reasonable to me. 

If medium becomes the default, do we still need a medium option, or would it be cleaner to omit that?

ciampo commented:
> * set `size="medium"` to be the default value.

Setting a preset size as default will potentially visually break existing usages of `Modal`, since suddenly the modal size would change to the new default value.

The safer approach would be to leave the default value as `undefined`.

jameskoster commented:
That's true, but isn't part of this effort to establish more consistent modal usage? At the moment there is a tendency to apply custom widths to each modal ad hoc, and without a more opinionated default I'd be a little anxious about that continuing. 

FWIW I wasn't suggesting automatically setting the `size` prop, just updating this css: https://github.com/WordPress/gutenberg/blob/878af2b1038bf49da8674ffd45fd84a974ac0570/packages/components/src/modal/style.scss#L38

So if a modal is already applying a custom size, it shouldn't be affected. But any new instances would effectively be `medium` rather than the current ~`100%` (which isn't very practical).

ciampo commented:
I believe that, in order to ensure ease of composition, each component should only take ownership of its bounding box, ie. it should not set its own size and margins. And therefore, if I were to build this component from scratch, I wouldn't add _any_ width/height in its default state, and grow as its content requires it.

But I understand that the component already currently comes with a set `max-width` and there is value in ensuring consistency by exposing a `size` prop.

> isn't part of this effort to establish more consistent modal usage? At the moment there is a tendency to apply custom widths to each modal ad hoc, and without a more opinionated default I'd be a little anxious about that continuing.

Having an additional property to pick a preset size from is a convenient way for folks to pick a size without much effort (compared to writing custom styles), and in a way that is consistent across the product.

> FWIW I wasn't suggesting automatically setting the `size` prop, just updating this css:
> 
> https://github.com/WordPress/gutenberg/blob/878af2b1038bf49da8674ffd45fd84a974ac0570/packages/components/src/modal/style.scss#L38
> 
> So if a modal is already applying a custom size, it shouldn't be affected. But any new instances would effectively be `medium` rather than the current ~`100%` (which isn't very practical).

I see what you mean. While this isn't as disruptive as adding `size='medium'` as default, it will still definitely introduce some level of disruption. But if you believe that it's worth to trade some disruption in exchange for more consistency (and a better default), then feel free to go ahead with the new proposed `max-width`.

chad1008 committed:
rename `contentWdith` to `size`

chad1008 committed:
extract widths to `base-styles` vars

chad1008 committed:
set medium default size

chad1008 commented:
> If medium becomes the default, do we still need a medium option, or would it be cleaner to omit that?

I had the exact same thought, but @ciampo pointed out in a separate convo that setting a default in the CSS will constrain the `max-width`, but still allow for smaller `Modal`s based on the size of their contents (down to the `min-width` at least). The actual `size="medium"` option actually forces that width, regardless of how small the contents are. So they do serve different purposes.

Granted, the current `min-width` is 350px, only 34px below the proposed default, so that flexibility likely won't come into play too often in practice. 🤔 

I've applied a medium default in the CSS for now, but if we decide against it, or if we feel like `size='medium'` is redundant I'm happy to make those adjustments'.

I've also abstracted the widths to variables in `base-styles` as suggested previously. Is `$modal-width-...`, etc., a set of names everyone is happy with? I thought of using `$dialog-width-...` to avoid being overly specific to `Modal`, but I was kind of on the fence.

jameskoster commented:
> But if you believe that it's worth to trade some disruption in exchange for more consistency (and a better default), then feel free to go ahead with the new proposed max-width.

It's hard to say without knowing how much disruption will be caused. Maybe we play it safe and leave the current max-width (sorry @chad1008 🙈). Like you said the new prop will hopefully encourage more responsible usage.

> I've also abstracted the widths to variables in base-styles as suggested previously. Is $modal-width-..., etc., a set of names everyone is happy with? I thought of using $dialog-width-... 

I was thinking more generic than that; `$container-width-...`, or maybe even just `$width-...`. @SaxonF had you thought about width variables at all relating to https://github.com/WordPress/gutenberg/issues/53322?

ciampo commented:
> It's hard to say without knowing how much disruption will be caused. Maybe we play it safe and leave the current max-width (sorry @chad1008 🙈). Like you said the new prop will hopefully encourage more responsible usage.

Alright, so to recap:

- we leave the current `max-width` as-is
- we introduce the new `size` prop, with a default value of `undefined` and possible values of `'small' | 'medium' | 'large' | 'fill'`,
- the `'fill'` option replaces the `isFullScreen` prop. As follow-up tasks, we should aim at:
  - Replace all usages of `isFullScreen` with `size="fill"`
  - Deprecate the `isFullScreen` prop
- As another follow-up, we should see if the new `size` prop can be used instead of the changes introduced in https://github.com/WordPress/gutenberg/pull/54518

 

> I was thinking more generic than that; `$container-width-...`, or maybe even just `$width-...`. @SaxonF had you thought about width variables at all relating to #53322?

I guess this conversation can happen in parallel from the changes in this PR. We can always refactor code internal to `Modal` since it wouldn't affect the consumers of the component.

chad1008 committed:
update CHANGELOG

chad1008 committed:
Revert "set medium default size"

This reverts commit ee8994413556f44b8604a079e17bce22c0a272e2.

chad1008 commented:
Excellent, thanks all! I've removed the new default width, which should bring the PR into alignment with @ciampo's helpful recap above.

I left the variable names as-is (`$modal-width-...`) for now, but will update them to a more generic name once we settle on one. In this PR if that happens before it's approved/merged, or as a followup if this merges first.

Rebased and updated CHANGELOG ✅

SaxonF commented:
> I was thinking more generic than that; $container-width-..., or maybe even just $width-.... @SaxonF had you thought about width variables at all relating to https://github.com/WordPress/gutenberg/issues/53322?

Yes we should work towards a set of standard width values as part of a scale. Once we get [colours](https://github.com/WordPress/gutenberg/pull/54541) in we can start working on other token types.

ciampo commented on packages/components/src/modal/README.md:
```diff
@@ -221,6 +221,14 @@ This property when set to `true` will render a full screen modal.
 -   Required: No
 -   Default: `false`
 
+#### `size`: `'small' | 'medium' | 'large'`
```
```suggestion
#### `size`: `'small' | 'medium' | 'large' | 'fill'`
```

jameskoster reviewed (approved):
Cool, I think we're good to go with this one. We can assign values from the generic container width scale to the `$modal-width...` variables when we get there if needs be.

ciampo commented on packages/components/src/modal/index.tsx:
```diff
@@ -228,7 +231,11 @@ function UnforwardedModal(
 						className,
 						{
-							'is-full-screen': isFullScreen,
+							'is-full-screen': isFullScreen || size === 'fill',
+						},
+						{
+							[ sizeClass ]:
+								! isFullScreen && size && size !== 'fill',
```
Let's put all the logic for the size classname in one place

```diff
diff --git a/packages/components/src/modal/index.tsx b/packages/components/src/modal/index.tsx
index 24d44b1a01..8b2916d5d8 100644
--- a/packages/components/src/modal/index.tsx
+++ b/packages/components/src/modal/index.tsx
@@ -98,7 +98,12 @@ function UnforwardedModal(
 	const [ hasScrolledContent, setHasScrolledContent ] = useState( false );
 	const [ hasScrollableContent, setHasScrollableContent ] = useState( false );
 
-	const sizeClass = size && size !== 'fill' ? `has-size-${ size }` : '';
+	let sizeClass;
+	if ( isFullScreen || size === 'fill' ) {
+		sizeClass = 'is-full-screen';
+	} else if ( size ) {
+		sizeClass = `has-size-${ size }`;
+	}
 
 	// Determines whether the Modal content is scrollable and updates the state.
 	const isContentScrollable = useCallback( () => {
@@ -229,14 +234,8 @@ function UnforwardedModal(
 				<div
 					className={ classnames(
 						'components-modal__frame',
-						className,
-						{
-							'is-full-screen': isFullScreen || size === 'fill',
-						},
-						{
-							[ sizeClass ]:
-								! isFullScreen && size && size !== 'fill',
-						}
+						sizeClass,
+						className
 					) }
 					style={ style }
 					ref={ useMergeRefs( [

```

chad1008 replied:
👍 d630574004401ae645cb8f764a7ce727dd367ca5

ciampo commented on packages/components/src/modal/README.md:
```diff
@@ -221,6 +221,14 @@ This property when set to `true` will render a full screen modal.
 
+#### `size`: `'small' | 'medium' | 'large'`
+
+If this property is added it will cause the modal to render at a preset width, or expand to fill the screen.
```
Let's add a note (both in the README and in the types JSDocs) that the `isFullScreen` prop will take precedence over this prop when set.

ciampo reviewed (approved):
LGTM 🚀 

Let a couple of comments, but this PR is good to be merged after they've been addressed.

We can tweak the internal width variables at a later point.

chad1008 committed:
include `fill` option in README

Co-authored-by: Marco Ciampini <marco.ciampo@gmail.com>

chad1008 committed:
consolidate sizing logic

chad1008 committed:
add note highlighting `isFullScreen` taking precedence

mirka commented:
@chad1008 I see Marco added a "Needs Dev Note" label, but I'm not sure what the intent was. I'm guessing it was just to inform consumers of this new `size` prop. However from this [later thread](https://github.com/WordPress/gutenberg/pull/55130#discussion_r1351732669), I was under the impression that we are not quite ready to advertise and encourage wide adoption of these size presets in their current state. Is that correct? Would you agree that we should probably refrain from a Dev Note here?

chad1008 commented:
@mirka Yes, the endgame behavior of the new `size` prop was kind of up in the air, and while we're using it internally it's probably best that that we don't encourage wider adoption. If a future change adds height constraints to this prop it could negatively impact existing uses we aren't aware of. With that in mind, I'd say holding off on a Dev Note makes sense.
